docs/DeveloperGuide.md

```diff
@@ -321,6 +321,35 @@ Below is a sequence diagram that shows a scenario where a suggestion is provided
 
 ![Interactions inside logic component and Algo component for Didyoumean feature](images/SuggestionSequenceDiagram.png)
 
+### Add Note feature
+
+#### Implementation
+
+*bookmark* allows Users to add notes to a book.
+
+This feature is facilitated mainly by `AddNoteCommandParser` and `AddCommand`.
+
+![Classes involved in the AddNoteCommand](images/NoteClassDiagram.png)
+
+`LogicManager#execute()` (not shown in the diagram above) handles the command word to create `AddNoteCommandParser` to parse the remaining inputs.
+`AddNoteCommandParser#parse()` tokenizes each prefix to create a `Note` object. This `Note` object will be
+passed as a parameter to create a `AddNoteCommand` that will be returned to `LogicManager`.<br>
+*If there are missing or invalid prefixes, an exception will be thrown with a message to the User.*
+
+`LogicManager#execute()` will call `AddNoteCommand#execute()` to add the `Book` with the added `Note` into
+the `Model` and return a `CommandResult` as a feedback to the user.
+*If there is an existing note with the same title and text, an exception will be thrown with a message to the User*
+
+Below is an activity diagram which illustrates the flow of events for adding a book
+
+![Add note command flow of execution](images/NoteActivityDiagram.png)
```

Comment:
I'm not too sure if we should include a conditional to check valid index in the activity diagram, because this diagram doesn't show that an error would be thrown.